### PR TITLE
[Google Blockly] Refactor CDO miniToolbox in block_utils

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR refactors the implementation of the mini toolbox in `block_utils.js` for CDO labs. It moves the logic for the mini toolbox to `cdoBlockly/spritelabBlocks.js` in the new `customBlocks/cdoBlockly` directory. There is a corresponding `googleBlockly/spritelabBlocks.js` in the new `customBlocks/googleBlockly` directory which is currently a placeholder. This allows us to not have to include new branching logic within `blockly_utils.js` depending on the Blockly version. 

There is no functionality change as a result of this PR. This is prep work for the implementation of the mini toolbox for Google Blockly labs that will be following up.

## Links

[jira](https://codedotorg.atlassian.net/browse/SL-992)

## Testing story
I tested locally to make sure functionality for the mini toolbox flyout did not change for CDO version of Sprite lab. 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/block_utils.js [hunk 1]
```diff
@@ -1063,67 +1063,18 @@ exports.createJsWrapperBlockCreator = function (
           this.setPreviousStatement(true);
         }
 
+        // Boolean constant to store when we show mini-toolbox.
         // Use window.appOptions, not global appOptions, because the levelbuilder
         // block page doesn't have appOptions, but we *do* want to show the mini-toolbox
-        // there
-        if (
+        // there.
+        const showMiniToolbox =
```
Comment (mikeharv) [inline apps/src/block_utils.js:1070]: Nice idea. This will help to simplify logic for Google Blockly fields that need to be created later on.

**Review by mikeharv — APPROVED** [review]
Thanks for moving this logic! It looks great. 

Quick note/clarification: Your PR description states that there are no functional changes here. While this is true for pretty much all of our users, anyone using `?blocklyVersion=Google` to for Sprite Lab to use Google Blockly will now see the event blocks render without errors. This is a great improvement!

CDO Blockly baseline:
image

Google Blockly on staging:
image

Google Blockly on this branch:
image

It also means you'll be able to open the `Events` category of any Sprite Lab toolbox in Google Blockly without encountering errors! :tada:

Staging:
image

Your branch:
image


The next steps will be to implement your [toggle field](https://github.com/code-dot-org/code-dot-org/pull/52708) and the [flyout field](https://github.com/code-dot-org/code-dot-org/pull/52711) so that we can the full functionality of the "mini-toolbox" working.

File: apps/src/blockly/customBlocks/cdoBlockly/spritelabBlocks.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+export const blocks = {
```
Comment (molly-moen) [inline apps/src/blockly/customBlocks/cdoBlockly/spritelabBlocks.js:3]: Can you add a top level comment explaining the purpose of this file? Similarly for the other new files.

**Review by molly-moen — APPROVED** [review]
Looks good to me, as a blockly newbie I have a request for more comments :)

File: apps/src/blockly/customBlocks/cdoBlockly/spritelabBlocks.js (hunk 2 above)
Comment (fisher-alice) [reply apps/src/blockly/customBlocks/cdoBlockly/spritelabBlocks.js:3]: Sure! Will do.

**Pull request merged**